draft-dss-star.md

```diff
@@ -101,200 +121,640 @@ In this document we describe one such system, namely Distributed Secret Sharing
+Clients are also configured with an Anonymizing Proxy that clients can use to send
+proxy reports to the Aggregation Server. The exact type of proxy is not specified here.
+See {{proxy-options}} for more details.
+
+### Reporting Protocol
+
+This reporting protocol works as follows. First, the client stretches `rand` into three values
+`key_seed`, `share_coins`, and `tag`, and additionally derives an KCAEAD key and nonce
+from `key_seed`.
+
+```
+// Randomness derivation
+rand_prk = Extract(nil, rand)
+key_seed = Expand(rand_prk, "key_seed", 16)
+share_coins = Expand(rand_prk, "share_coins", 16)
+tag = Expand(rand_prk, "tag", 16)
+
+// Symmetric encryption key derivation
+key_prk = Extract(nil, key_seed)
+key = Expand(key_prk, "key", Nk)
+nonce = Expand(key_prk, "nonce", Nn)
+```
+
+The client then generates a secret share of `key_seed` using `share_coins` as randomness as follows:
+
+```
+rand_share = Share(REPORT_THRESHOLD, TBD, key_seed, share_coins, nil)
+```
+
+[[OPEN ISSUE: what should N be for the TSS scheme?]]
+
+The client then encrypts `msg` and `aux` using the KCAEAD key and nonce as follows:
+
+```
+report_data = msg || aux
+encrypted_report = Seal(key, nonce, nil, report_data)
+```
+
+Finally, the client constructs a report consisting of `encrypted_report`, `rand_share`,
+and `tag`, and sends this to the Anonymizing Server in the subsequent epoch, i.e., after
+the Randomness Server has rotated its OPRF key.
+
+[[OPEN ISSUE: how does the client know when this rotation happens?]]
+
+```
+struct {
+  opaque encrypted_report<1..2^16-1>;
+  opaque rand_share[Nshare];
+  opaque tag[16];
+} Report;
+```
+
+Specifically, Clients send a Report to the Aggregation Server using an HTTP POST message
+with content type "message/star-report". An example message is below.
+
+~~~
+:method = POST
+:scheme = https
+:authority = aggregator.example
+:path = /
+content-type = message/star-report
+content-length = <Length of body>
+
+<Bytes containing a Report>
+~~~
+
+This message is sent to the Aggregation Server through the Anonymizing Proxy. See {{proxy-options}}
+for different types of proxy options.
+
+## Aggregation Phase
+
+Aggregation is the final phase of STAR. It happens offline and does not require any
+communication between different STAR entities. It proceeds as follows. First, the
+Aggregation Server groups reports together based on their `tag` value. Let `report_set`
+denote a set of at least REPORT_THRESHOLD reports that have a matching `tag` value.
+
+Given this set, the Aggregation Server begins by running the secret share recovery algoritm
+as follows:
+
+```
+key_seed = Recover(report_set)
+```
+
+If this fails, the Aggregation Server chooses a new candidate report share set and
+reruns the aggregation process.
+
+[[OPEN ISSUE: how does the server choose new candidate sets when share recovery fails?]]
```

Comment:
I think robust secret-sharing solves this, e.g. https://www.cwi.nl/system/files/Fehr.pdf.